src/common/utilities.js

```diff
@@ -1,3 +1,4 @@
+/* eslint-disable node/no-unsupported-features/node-builtins */
```

Comment:
Please disable in `.eslintrc.js` `overrides`. Anything environment-related should be set in `.eslintrc.js` `overrides` where we can specify the environment without cluttering our files (if these are browser files, we don't need such node rules reporting in *any* of our browser files).

(The reason I apply `sauron-node` instead of `sauron` though, which I believe is the reason we are seeing this rule added,  is IIRC, because of an extremely helpful "Node" linting rule which requires that all imports are present in `package.json` dependencies unless the file is within `.npmignore`. Otherwise, things may end up working in our testing environment, but the package would not be available to users. But `node-builtins` at least is not a rule we need to enforce in browser-only files.)

Reply:
OK, I was a little more "disrupting" about lint rules. 
First, I wanted to start with lint rule on our source code in priority (i.e. I only lint the src folder. This means that .md files are not currently checked)
Second, I added the eslint:recommended" rules. 
Third, I turned most rules from "off" into "warn" so we can have a good idea of the work to make our code perfect but some of them are really broken everywhere so I turned them back to "off" at least for now.
Third, I removed the max-warning  0. In fact, this flag is counter productive because this push developer to disable rules so the build passes. Now, the build passes but we are reminded that the code should be reworked to comply with all rules.

Reply:
Before I reply re: linting, one off topic FYI--though I won't it add to this PR in case it causes any problems... I'd like to add https://github.com/hkdobrev/run-if-changed to the project, which should ensure that whenever `package.json` (or `package-lock.json` perhaps) is changed, the following will occur when we make such changes ourselves or merge the others changes:

1. When we merge each other's changes, we don't need to always remember to re-run `npm install` to ensure we are working against the same dependencies.
2. `license-badges` is run immediately after new versions are updated (rather than at publish time as it currently works), ensuring our license badge summaries on the README are up to date.
3. I'm working on a project, [integrity-matters](https://github.com/brettz9/integrity-matters) to facilitate auto-updating of CDN URLs and adding `integrity` attributes which ensure users are only loading scripts from CDNs which match the version we have in npm. If a new version is installed locally, `run-if-changed` could run this tool--after I may get it fully tested/covered as I'm working on now--to allow us to add CDN URLs, e.g., for jquery, and auto-update such URLs into the future and ensure they are pointing to the same version as we have installed locally, and then, after confirming, applying its changed `integrity` attribute hash also (avoiding the need for us to visit the download page for jquery, etc. each time there is an update, to find and then manually add the updated integrity values and new versioned URL).